### PR TITLE
[PART 2] TF-3002 Search more filters

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#3002

## Resolved

- Mobile:

[demo-mobile.webm](https://github.com/user-attachments/assets/de77d942-a97e-4aeb-8943-675cda0b0476)

- Web responsive:

https://github.com/user-attachments/assets/beb961fb-9961-4eb1-985f-6cee40496c97

### CODE REVIEW

**Review by tddang-linagora — APPROVED** [review]

**Comment by hoangdat** [comment]
- please change the base to: [search-improvement-sprint-28](https://github.com/linagora/tmail-flutter/tree/search-improvement-sprint-28) and rebase @dab246

**Comment by dab246** [comment]
> * please change the base to: [search-improvement-sprint-28](https://github.com/linagora/tmail-flutter/tree/search-improvement-sprint-28) and rebase @dab246

Done. @hoangdat

**Comment by hoangdat** [comment]
- [ ] after search with keyword and other filters, I can not clear keyword to search only with filters

https://github.com/user-attachments/assets/9740be0c-496e-40c1-95be-a1678fc8679f

**Comment by dab246** [comment]
> * [ ]  after search with keyword and other filters, I can not clear keyword to search only with filters
> 
>  Screen.Recording.2024-10-03.at.10.55.53.mov

Clear filter only deletes text and gives suggestions. You can only search after clicking enter or pressing the search button.

**Comment by hoangdat** [comment]
- [ ] [MOBILE] when user select contact suggestion, search filter also apply this contact in both `text` and `from`. 
- Expected: only `from`

https://github.com/user-attachments/assets/fc771f44-5af6-4b73-a7a2-821f183f1082

**Comment by dab246** [comment]
> * [x]  [MOBILE] when user select contact suggestion, search filter also apply this contact in both `text` and `from`.
> * Expected: only `from`
> 
>  Screen.Recording.2024-10-03.at.11.05.41.mov

Fixed


https://github.com/user-attachments/assets/1800587f-1596-49a9-94cb-45b4210f1ce6

**Comment by hoangdat** [comment]
- [ ] `no result` screen still show during search is in progress
[Screen_recording_20241003_113618.webm](https://github.com/user-attachments/assets/2818f49e-7084-414e-8dc1-efdca31e98ea)

**Comment by hoangdat** [comment]
- [ ] write test to ensure search filter send to interactor is exactly which user changed

**Comment by dab246** [comment]
> * [x]  `no result` screen still show during search is in progress
>   [Screen_recording_20241003_113618.webm](https://github.com/user-attachments/assets/2818f49e-7084-414e-8dc1-efdca31e98ea)

Fixed.

[Screen_recording_20241003_144651.webm](https://github.com/user-attachments/assets/b7c424af-2575-4b4a-8578-e8645aa14ef2)

**Comment by dab246** [comment]
> * [x]  write test to ensure search filter send to interactor is exactly which user changed

Updated

File: lib/features/search/email/presentation/search_email_controller.dart [hunk 1]
```diff
@@ -172,6 +191,22 @@ class SearchEmailController extends BaseController
     }
   }
 
+  @override
+  void handleUrgentExceptionOnMobile({Failure? failure, Exception? exception}) {
+    super.handleUrgentExceptionOnMobile(failure: failure, exception: exception);
+    if (failure is SearchEmailFailure) {
+      _searchEmailsFailure(failure);
+    }
```
Comment (hoangdat) [inline lib/features/search/email/presentation/search_email_controller.dart:199]: can you demo this case?

**Comment by hoangdat** [comment]
- [ ] clear text in advanced search, but text in search bar is not deleted too (filter in request is well updated)

https://github.com/user-attachments/assets/d63d613a-4fca-42c9-8330-7a7af15d6707

**Review by hoangdat — APPROVED** [review]

**Pull request merged**